### ISSUE TITLE
Get rid of all hidden method forbidden calls on higher API (dark greylist, reflection)

### ISSUE THREAD

**Issue by courville** [nova-video-player/aos-AVP#8]
FileCoreLibrary/src/com/archos/filecorelibrary/ExtStorageManager.java uses some hidden method calls and is reported in debug mode via a warning popup stating "Detected problems with API compatibility".

```
08-27 17:48:20.646 5979-5979/org.courville.nova W/.courville.nov: Accessing hidden method Ldalvik/system/CloseGuard;->get()Ldalvik/system/CloseGuard; (light greylist, reflection)
    Accessing hidden method Ldalvik/system/CloseGuard;->open(Ljava/lang/String;)V (light greylist, reflection)
    Accessing hidden method Ldalvik/system/CloseGuard;->warnIfOpen()V (light greylist, reflection)
08-27 17:48:21.317 5979-5979/org.courville.nova W/.courville.nov: Accessing hidden method Landroid/os/storage/StorageEventListener;-><init>()V (light greylist, linking)
    Accessing hidden method Landroid/os/storage/StorageManager;->registerListener(Landroid/os/storage/StorageEventListener;)V (light greylist, reflection)
08-27 17:48:21.319 5979-5979/org.courville.nova W/.courville.nov: Accessing hidden method Landroid/os/ServiceManager;->getService(Ljava/lang/String;)Landroid/os/IBinder; (light greylist, reflection)
    Accessing hidden method Landroid/os/storage/IStorageManager$Stub;->asInterface(Landroid/os/IBinder;)Landroid/os/storage/IStorageManager; (light greylist, reflection)
08-27 17:48:21.320 5979-5979/org.courville.nova W/.courville.nov: Accessing hidden method Landroid/os/storage/StorageVolume;->getPath()Ljava/lang/String; (light greylist, reflection)
    Accessing hidden method Landroid/os/storage/VolumeInfo;->getFsUuid()Ljava/lang/String; (light greylist, reflection)
    Accessing hidden method Landroid/os/storage/VolumeInfo;->isMountedReadable()Z (light greylist, reflection)
    Accessing hidden method Landroid/os/storage/VolumeInfo;->getDisk()Landroid/os/storage/DiskInfo; (light greylist, reflection)
    Accessing hidden method Landroid/os/storage/VolumeInfo;->getPath()Ljava/io/File; (light greylist, reflection)
    Accessing hidden method Landroid/os/storage/DiskInfo;->isSd()Z (light greylist, reflection)
    Accessing hidden method Landroid/os/storage/DiskInfo;->isUsb()Z (light greylist, reflection)
    Accessing hidden field Landroid/os/storage/VolumeInfo;->type:I (light greylist, reflection)
    Accessing hidden method Landroid/os/storage/IStorageManager;->getVolumes(I)[Landroid/os/storage/VolumeInfo; (dark greylist, reflection)
08-27 17:48:21.674 5979-5979/org.courville.nova W/.courville.nov: Accessing hidden method Landroid/view/View;->computeFitSystemWindows(Landroid/graphics/Rect;Landroid/graphics/Rect;)Z (light greylist, reflection)
08-27 17:48:21.689 5979-5979/org.courville.nova W/.courville.nov: Accessing hidden method Landroid/view/ViewGroup;->makeOptionalFitsSystemWindows()V (light greylist, reflection)
08-27 17:48:22.200 5979-5979/org.courville.nova W/.courville.nov: Accessing hidden field Landroid/view/animation/Animation;->mListener:Landroid/view/animation/Animation$AnimationListener; (light greylist, reflection)
08-27 17:48:22.838 5979-5979/org.courville.nova W/.courville.nov: Accessing hidden method Landroid/widget/AutoCompleteTextView;->doBeforeTextChanged()V (light greylist, reflection)
    Accessing hidden method Landroid/widget/AutoCompleteTextView;->doAfterTextChanged()V (light greylist, reflection)
    Accessing hidden method Landroid/widget/AutoCompleteTextView;->ensureImeVisible(Z)V (light greylist, reflection)
```

**Comment by courville** [comment]
ExtStorageManager.java and ExtStorageReceiver.java needs to be changed in order not to access hidden methods from android.os.storage.StorageEventListener to detect new mounted Volumes on Android.
One could assume that listening to MEDIA intents (no need for dynamic broadcast receiver registration since these intents are part of broadcast exceptions) namely
android.intent.action.MEDIA_{MOUNTED,UNMOUNTED,EJECT,BAD_REMOVAL} is enough to detect insertion of sdcard or usb mass storage devices but sadly this is not the case.
Indeed on Nexus/Pixel devices (unlikely Samsung phones) usb key insertion does not trigger Intent.ACTION_MEDIA_MOUNTED. Thus in order to browse usb mass storage, android.hardware.usb.action.USB_DEVICE_{ATTACHED,DETACHED} must be monitored via storageManager.getStorageVolumes() in case of an usbInterface.getInterfaceClass() == UsbConstants.USB_CLASS_MASS_STORAGE.

**Comment by courville** [comment]
OK... turns out that on nexus devices for external usb mass storage, the MEDIA_MOUNTED intent is not received because the device is not mounted per say and that the access is done in user space.
Thus the only todo is to remove the introspections and everything "should work"(tm).

**Comment by courville** [comment]
For reference sake, https://www.nplix.com/how-to-list-all-sdcard-android-storage-manager/ is interesting to read.

**Comment by courville** [comment]
The first issue to treat is the dark greylist because https://developer.android.com/about/versions/pie/restrictions-non-sdk-interfaces says that it will prevent to target API 28 i.e. android/os/storage/IStorageManager;->getVolumes
Idea: preserve existing code and only on P and above use instead StorageManager.getStorageVolumes.

**Comment by courville** [comment]
Only on P and above use instead StorageManager.getStorageVolumes.

**Comment by courville** [comment]
In the future to monitor greylisting status check https://android.googlesource.com/platform/frameworks/base/+/android-9.0.0_r21/config/
https://android.googlesource.com/platform/prebuilts/runtime/+/master/appcompat

**Comment by courville** [comment]
veridex tool https://android.googlesource.com/platform/prebuilts/runtime/+/master/appcompat reports via `./appcompat.sh --dex-file=Video*release.apk`

```
NOTE: appcompat.sh is still under development. It can report
API uses that do not execute at runtime, and reflection uses
that do not exist. It can also miss on reflection uses.
#1: Linking greylist Lorg/json/JSONObject;->keySet()Ljava/util/Set; use(s):
       Lorg/json/CookieList;->toString(Lorg/json/JSONObject;)Ljava/lang/String;
       Lorg/json/HTTP;->toString(Lorg/json/JSONObject;)Ljava/lang/String;
       Lorg/json/JSONML;->toString(Lorg/json/JSONArray;)Ljava/lang/String;
       Lorg/json/JSONML;->toString(Lorg/json/JSONObject;)Ljava/lang/String;
       Lorg/json/JSONObject;->getNames(Lorg/json/JSONObject;)[Ljava/lang/String;
       Lorg/json/JSONObject;->keys()Ljava/util/Iterator;
       Lorg/json/JSONObject;->similar(Ljava/lang/Object;)Z (2 occurrences)
       Lorg/json/Property;->toProperties(Lorg/json/JSONObject;)Ljava/util/Properties;
       Lorg/json/XML;->toString(Ljava/lang/Object;Ljava/lang/String;)Ljava/lang/String;

#2: Reflection greylist Landroid/animation/LayoutTransition;->cancel use(s):
       Landroid/support/transition/ViewGroupUtilsApi14;->cancelLayoutTransition(Landroid/animation/LayoutTransition;)V

#3: Reflection greylist Landroid/app/ActionBar;->setShowHideAnimationEnabled use(s):
       Lcom/archos/mediacenter/video/player/PlayerActivity;->onCreate(Landroid/os/Bundle;)V

#4: Reflection greylist Landroid/content/res/Resources;->mResourcesImpl use(s):
       Landroid/support/v7/app/ResourcesFlusher;->flushNougats(Landroid/content/res/Resources;)Z

#5: Reflection greylist-max-o Landroid/database/sqlite/SQLiteDatabase;->addCustomFunction use(s):
       Lcom/archos/mediaprovider/SQLiteDbProxy;->invoke(Ljava/lang/Class;Landroid/database/sqlite/SQLiteDatabase;Ljava/lang/String;ILjava/lang/Object;)Z

#6: Reflection greylist Landroid/graphics/FontFamily;->abortCreation use(s):
       Landroid/support/v4/graphics/TypefaceCompatApi26Impl;-><clinit>()V

#7: Reflection greylist Landroid/graphics/FontFamily;->addFontFromAssetManager use(s):
       Landroid/support/v4/graphics/TypefaceCompatApi26Impl;-><clinit>()V

#8: Reflection greylist Landroid/graphics/FontFamily;->addFontFromBuffer use(s):
       Landroid/support/v4/graphics/TypefaceCompatApi26Impl;-><clinit>()V

#9: Reflection greylist Landroid/graphics/FontFamily;->freeze use(s):
       Landroid/support/v4/graphics/TypefaceCompatApi26Impl;-><clinit>()V

#10: Reflection blacklist Landroid/graphics/Typeface;->createFromFamiliesWithDefault use(s):
       Landroid/support/v4/graphics/TypefaceCompatApi24Impl;-><clinit>()V
       Landroid/support/v4/graphics/TypefaceCompatApi26Impl;-><clinit>()V

#11: Reflection blacklist Landroid/graphics/drawable/Drawable;->getOpticalInsets use(s):
       Landroid/support/v7/widget/DrawableUtils;->getOpticalBounds(Landroid/graphics/drawable/Drawable;)Landroid/graphics/Rect;

#12: Reflection greylist Landroid/graphics/drawable/Drawable;->isProjected use(s):
       Landroid/support/v4/graphics/drawable/WrappedDrawableApi21;->findAndCacheIsProjectedDrawableMethod()V

#13: Reflection greylist Landroid/media/AudioAttributes;->toLegacyStreamType use(s):
       Landroid/support/v4/media/AudioAttributesCompatApi21;->toLegacyStreamType(Landroid/support/v4/media/AudioAttributesCompatApi21$Wrapper;)I

#14: Reflection greylist Landroid/media/MediaPlayer;->setDataSource use(s):
       Lcom/archos/medialib/AndroidMediaPlayer;->setDataSource2(Ljava/lang/String;Ljava/util/Map;)V

#15: Reflection greylist Landroid/media/session/MediaSession;->getCallingPackage use(s):
       Landroid/support/v4/media/session/MediaSessionCompatApi24;->getCallingPackage(Ljava/lang/Object;)Ljava/lang/String;

#16: Reflection greylist Landroid/os/Bundle;->getIBinder use(s):
       Landroid/support/v4/app/BundleCompat$BundleCompatBaseImpl;->getBinder(Landroid/os/Bundle;Ljava/lang/String;)Landroid/os/IBinder;

#17: Reflection greylist Landroid/os/Bundle;->putIBinder use(s):
       Landroid/support/v4/app/BundleCompat$BundleCompatBaseImpl;->putBinder(Landroid/os/Bundle;Ljava/lang/String;Landroid/os/IBinder;)V

#18: Reflection greylist Landroid/os/ServiceManager;->getService use(s):
       Lcom/archos/filecorelibrary/ExtStorageManager;->getVolumeState(Ljava/lang/String;)Ljava/lang/String;
       Lcom/archos/filecorelibrary/ExtStorageManager;->updateAllVolumes()V

#19: Reflection greylist Landroid/os/SystemProperties;->set use(s):
       Lcom/archos/environment/SystemPropertiesProxy;->set(Ljava/lang/String;Ljava/lang/String;)V

#20: Reflection greylist Landroid/os/storage/DiskInfo;->isSd use(s):
       Lcom/archos/filecorelibrary/ExtStorageManager;->updateAllVolumes()V

#21: Reflection greylist Landroid/os/storage/DiskInfo;->isUsb use(s):
       Lcom/archos/filecorelibrary/ExtStorageManager;->updateAllVolumes()V

#22: Reflection greylist Landroid/os/storage/IStorageManager$Stub;->asInterface use(s):
       Lcom/archos/filecorelibrary/ExtStorageManager;->updateAllVolumes()V

#23: Reflection greylist-max-o Landroid/os/storage/IStorageManager;->getVolumes use(s):
       Lcom/archos/filecorelibrary/ExtStorageManager;->updateAllVolumes()V

#24: Reflection greylist Landroid/os/storage/StorageManager;->getVolumePaths use(s):
       Lcom/archos/filecorelibrary/ExtStorageManager;->updateAllVolumes()V
       Lcom/archos/mediaprovider/VolumeState;->getVolumePaths(Landroid/os/storage/StorageManager;)[Ljava/lang/String;

#25: Reflection greylist Landroid/os/storage/StorageManager;->registerListener use(s):
       Lcom/archos/filecorelibrary/ExtStorageManager;-><init>()V

#26: Reflection greylist Landroid/os/storage/StorageVolume;->getPath use(s):
       Lcom/archos/filecorelibrary/ExtStorageManager;->updateAllVolumes()V

#27: Reflection greylist Landroid/os/storage/VolumeInfo;->getDisk use(s):
       Lcom/archos/filecorelibrary/ExtStorageManager;->updateAllVolumes()V

#28: Reflection greylist Landroid/os/storage/VolumeInfo;->getFsUuid use(s):
       Lcom/archos/filecorelibrary/ExtStorageManager;->updateAllVolumes()V

#29: Reflection greylist Landroid/os/storage/VolumeInfo;->getPath use(s):
       Lcom/archos/filecorelibrary/ExtStorageManager;->updateAllVolumes()V

#30: Reflection greylist Landroid/os/storage/VolumeInfo;->isMountedReadable use(s):
       Lcom/archos/filecorelibrary/ExtStorageManager;->updateAllVolumes()V

#31: Reflection greylist Landroid/os/storage/VolumeInfo;->type use(s):
       Lcom/archos/filecorelibrary/ExtStorageManager;->updateAllVolumes()V

#32: Reflection greylist Landroid/service/media/MediaBrowserService$Result;->mFlags use(s):
       Landroid/support/v4/media/MediaBrowserServiceCompatApi26;-><clinit>()V

#33: Reflection greylist Landroid/view/LayoutInflater;->mFactory2 use(s):
       Landroid/support/v4/view/LayoutInflaterCompat;->forceSetFactory2(Landroid/view/LayoutInflater;Landroid/view/LayoutInflater$Factory2;)V

#34: Reflection greylist Landroid/view/View;->computeFitSystemWindows use(s):
       Landroid/support/v7/widget/ViewUtils;-><clinit>()V

#35: Reflection greylist Landroid/view/View;->getTransitionAlpha use(s):
       Landroid/support/transition/ViewUtilsApi19;->fetchGetTransitionAlphaMethod()V

#36: Reflection greylist Landroid/view/View;->mAccessibilityDelegate use(s):
       Landroid/support/v4/view/ViewCompat$ViewCompatBaseImpl;->hasAccessibilityDelegate(Landroid/view/View;)Z

#37: Reflection greylist Landroid/view/View;->mMinHeight use(s):
       Landroid/support/v4/view/ViewCompat$ViewCompatBaseImpl;->getMinimumHeight(Landroid/view/View;)I

#38: Reflection greylist Landroid/view/View;->mMinWidth use(s):
       Landroid/support/v4/view/ViewCompat$ViewCompatBaseImpl;->getMinimumWidth(Landroid/view/View;)I

#39: Reflection greylist Landroid/view/View;->mRecreateDisplayList use(s):
       Landroid/support/v4/widget/SlidingPaneLayout$SlidingPanelLayoutImplJB;-><init>()V

#40: Reflection greylist Landroid/view/View;->mViewFlags use(s):
       Landroid/support/transition/ViewUtils;->fetchViewFlagsField()V

#41: Reflection greylist Landroid/view/View;->setAnimationMatrix use(s):
       Landroid/support/transition/ViewUtilsApi21;->fetchSetAnimationMatrix()V

#42: Reflection greylist Landroid/view/View;->setLeftTopRightBottom use(s):
       Landroid/support/transition/ViewUtilsApi22;->fetchSetLeftTopRightBottomMethod()V

#43: Reflection greylist Landroid/view/View;->setTransitionAlpha use(s):
       Landroid/support/transition/ViewUtilsApi19;->fetchSetTransitionAlphaMethod()V

#44: Reflection greylist Landroid/view/View;->transformMatrixToGlobal use(s):
       Landroid/support/transition/ViewUtilsApi21;->fetchTransformMatrixToGlobalMethod()V

#45: Reflection greylist Landroid/view/View;->transformMatrixToLocal use(s):
       Landroid/support/transition/ViewUtilsApi21;->fetchTransformMatrixToLocalMethod()V

#46: Reflection greylist Landroid/view/ViewGroup;->suppressLayout use(s):
       Landroid/support/transition/ViewGroupUtilsApi18;->fetchSuppressLayoutMethod()V

#47: Reflection greylist Landroid/view/animation/Animation;->mListener use(s):
       Landroid/support/v4/app/FragmentManagerImpl;->getAnimationListener(Landroid/view/animation/Animation;)Landroid/view/animation/Animation$AnimationListener;

#48: Reflection greylist Landroid/widget/AbsListView;->mIsChildViewEnabled use(s):
       Landroid/support/v7/widget/DropDownListView;-><init>(Landroid/content/Context;Z)V

#49: Reflection greylist Landroid/widget/AutoCompleteTextView;->doAfterTextChanged use(s):
       Landroid/support/v7/widget/SearchView$AutoCompleteTextViewReflector;-><init>()V

#50: Reflection greylist Landroid/widget/AutoCompleteTextView;->doBeforeTextChanged use(s):
       Landroid/support/v7/widget/SearchView$AutoCompleteTextViewReflector;-><init>()V

#51: Reflection greylist Landroid/widget/AutoCompleteTextView;->ensureImeVisible use(s):
       Landroid/support/v7/widget/SearchView$AutoCompleteTextViewReflector;-><init>()V

#52: Reflection greylist Landroid/widget/CompoundButton;->mButtonDrawable use(s):
       Landroid/support/v4/widget/CompoundButtonCompat$CompoundButtonCompatBaseImpl;->getButtonDrawable(Landroid/widget/CompoundButton;)Landroid/graphics/drawable/Drawable;

#53: Reflection greylist Landroid/widget/ImageView;->animateTransform use(s):
       Landroid/support/transition/ImageViewUtilsApi21;->fetchAnimateTransformMethod()V

#54: Reflection greylist Landroid/widget/PopupWindow;->mOverlapAnchor use(s):
       Landroid/support/v4/widget/PopupWindowCompat$PopupWindowCompatApi21Impl;-><clinit>()V

#55: Reflection greylist Landroid/widget/PopupWindow;->setClipToScreenEnabled use(s):
       Landroid/support/v7/widget/ListPopupWindow;-><clinit>()V

#56: Reflection greylist Landroid/widget/PopupWindow;->setEpicenterBounds use(s):
       Landroid/support/v7/widget/ListPopupWindow;-><clinit>()V

#57: Reflection greylist Landroid/widget/PopupWindow;->setTouchModal use(s):
       Landroid/support/v7/widget/MenuPopupWindow;-><clinit>()V

#58: Reflection greylist Landroid/widget/TextView;->getHorizontallyScrolling use(s):
       Landroid/support/v7/widget/AppCompatTextViewAutoSizeHelper;->autoSizeText()V

#59: Reflection greylist Landroid/widget/TextView;->getLayoutAlignment use(s):
       Landroid/support/v7/widget/AppCompatTextViewAutoSizeHelper;->suggestedSizeFitsInSpace(ILandroid/graphics/RectF;)Z

#60: Reflection greylist Landroid/widget/TextView;->getTextDirectionHeuristic use(s):
       Landroid/support/v7/widget/AppCompatTextViewAutoSizeHelper;->createStaticLayoutForMeasuring(Ljava/lang/CharSequence;Landroid/text/Layout$Alignment;II)Landroid/text/StaticLayout;

#61: Reflection greylist Landroid/widget/TextView;->mMaxMode use(s):
       Landroid/support/v4/widget/TextViewCompat$TextViewCompatBaseImpl;->getMaxLines(Landroid/widget/TextView;)I

#62: Reflection greylist Landroid/widget/TextView;->mMaximum use(s):
       Landroid/support/v4/widget/TextViewCompat$TextViewCompatBaseImpl;->getMaxLines(Landroid/widget/TextView;)I
       Landroid/support/v4/widget/TextViewCompat$TextViewCompatBaseImpl;->getMaxLines(Landroid/widget/TextView;)I

#63: Reflection greylist Landroid/widget/TextView;->mMinMode use(s):
       Landroid/support/v4/widget/TextViewCompat$TextViewCompatBaseImpl;->getMinLines(Landroid/widget/TextView;)I

#64: Reflection greylist Landroid/widget/TextView;->mMinimum use(s):
       Landroid/support/v4/widget/TextViewCompat$TextViewCompatBaseImpl;->getMinLines(Landroid/widget/TextView;)I
       Landroid/support/v4/widget/TextViewCompat$TextViewCompatBaseImpl;->getMinLines(Landroid/widget/TextView;)I

#65: Reflection greylist Landroid/widget/TextView;->nullLayouts use(s):
       Landroid/support/v7/widget/AppCompatTextViewAutoSizeHelper;->setRawTextSize(F)V

#66: Reflection greylist Ldalvik/system/CloseGuard;->get use(s):
       Lokhttp3/internal/platform/AndroidPlatform$CloseGuard;->get()Lokhttp3/internal/platform/AndroidPlatform$CloseGuard;

#67: Reflection greylist Ldalvik/system/CloseGuard;->open use(s):
       Lokhttp3/internal/platform/AndroidPlatform$CloseGuard;->get()Lokhttp3/internal/platform/AndroidPlatform$CloseGuard;

#68: Reflection greylist Ldalvik/system/CloseGuard;->warnIfOpen use(s):
       Lokhttp3/internal/platform/AndroidPlatform$CloseGuard;->get()Lokhttp3/internal/platform/AndroidPlatform$CloseGuard;

#69: Reflection greylist Ljava/net/InetAddress;->holder use(s):
       Lorg/fourthline/cling/android/AndroidNetworkAddressFactory;->isUsableAddress(Ljava/net/NetworkInterface;Ljava/net/InetAddress;)Z

#70: Reflection greylist Llibcore/icu/ICU;->addLikelySubtags use(s):
       Landroid/support/v4/text/ICUCompat;-><clinit>()V

#71: Reflection greylist Lsun/misc/Unsafe;->allocateInstance use(s):
       Lcom/google/gson/internal/UnsafeAllocator;->create()Lcom/google/gson/internal/UnsafeAllocator;

#72: Reflection greylist Lsun/misc/Unsafe;->theUnsafe use(s):
       Lcom/google/gson/internal/UnsafeAllocator;->create()Lcom/google/gson/internal/UnsafeAllocator;

72 hidden API(s) used: 1 linked against, 71 through reflection
       68 in greylist
       2 in blacklist
       2 in greylist-max-o
       0 in greylist-max-p
To run an analysis that can give more reflection accesses, 
but could include false positives, pass the --imprecise flag. 
```

**Comment by courville** [comment]
tl;dr:
```
#5: Reflection greylist-max-o Landroid/database/sqlite/SQLiteDatabase;->addCustomFunction use(s):
       Lcom/archos/mediaprovider/SQLiteDbProxy;->invoke(Ljava/lang/Class;Landroid/database/sqlite/SQLiteDatabase;Ljava/lang/String;ILjava/lang/Object;)Z

#23: Reflection greylist-max-o Landroid/os/storage/IStorageManager;->getVolumes use(s):
       Lcom/archos/filecorelibrary/ExtStorageManager;->updateAllVolumes()V

#10: Reflection blacklist Landroid/graphics/Typeface;->createFromFamiliesWithDefault use(s):
       Landroid/support/v4/graphics/TypefaceCompatApi24Impl;-><clinit>()V
       Landroid/support/v4/graphics/TypefaceCompatApi26Impl;-><clinit>()V

#11: Reflection blacklist Landroid/graphics/drawable/Drawable;->getOpticalInsets use(s):
       Landroid/support/v7/widget/DrawableUtils;->getOpticalBounds(Landroid/graphics/drawable/Drawable;)Landroid/graphics/Rect;

```

**Comment by courville** [comment]
This should address the dark greylist issues https://github.com/nova-video-player/aos-MediaLib/commit/0c5a204972f98333c13b864cd6cf7cee8d840b5a https://github.com/nova-video-player/aos-FileCoreLibrary/commit/04be13c5bbcfccc3a9651743530c71052d3177ed

**Comment by courville** [comment]
It seems that P20 looses local storage video indexing with this patch.

**Comment by courville** [comment]
On api28 androidx branch running on Q we have forbidden getVolumes crashing:
```
Accessing hidden method Landroid/os/storage/IStorageManager;->getVolumes(I)[Landroid/os/storage/VolumeInfo; (greylist-max-o, reflection, denied)
```

**Comment by courville** [comment]
For API29 here are the next steps:
```
Accessing hidden method Landroid/widget/AutoCompleteTextView;->doBeforeTextChanged()V (greylist-max-p, reflection, allowed)
Accessing hidden method Landroid/widget/AutoCompleteTextView;->doAfterTextChanged()V (greylist-max-p, reflection, allowed)
Accessing hidden method Landroid/widget/AutoCompleteTextView;->ensureImeVisible(Z)V (greylist-max-p, reflection, allowed)
```
